### DISCUSSION TITLE
Compiled PropertyInfo Getter/Setter

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#58942] [Q&A]
Does anyone know of a good nuget package that can convert reflection-based getters and setters into high-speed actions and funcs?  I'm not even sure what to search for on nuget.org.

Thanks for the help!

**Comment by acaly** [comment]
Can you use `MethodInfo.CreateDelegate`? After binding to a delegate, reflection-based `MethodInfo` should be very fast to call.

**Reply by fernandozago** [reply]
Something Like:

```csharp
public class MyClassToBeReflected
    {
        private int ThePrivateProperty { get; set; } = 69;
    }


    class Program
    {
        private static readonly Func<MyClassToBeReflected, int> GetGetFunction =
            typeof(MyClassToBeReflected)
            .GetProperty("ThePrivateProperty", BindingFlags.NonPublic | BindingFlags.Instance)
            .GetGetMethod(true)!.CreateDelegate<Func<MyClassToBeReflected, int>>();

        private static readonly Action<MyClassToBeReflected, int> GetSetAction =
            typeof(MyClassToBeReflected)
            .GetProperty("ThePrivateProperty", BindingFlags.NonPublic | BindingFlags.Instance)
            .GetSetMethod(true)!.CreateDelegate<Action<MyClassToBeReflected, int>>();

        static void Main()
        {
            MyClassToBeReflected myInstance = new MyClassToBeReflected();
            int value = GetGetFunction(myInstance);
            Console.WriteLine($"Original Value: {value}");

            GetSetAction(myInstance, value + 1);
            value = GetGetFunction(myInstance);
            Console.WriteLine($"Changed Value: {value}");
            Console.ReadKey();
        }
    }
```

**Comment by msedi** [comment]
Can you extend your classes? If yes, as far as I have have read the Microsoft Community Toolkit now uses source generators to create the getters and setters (see: https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/3872). I haven't tested it but I assume source generators is the keyword you have to search for.

If the solutions you find do not fit it should be pretty easy (although not without work) to create your own source generator.
